db/versions/0010.yml

```diff
@@ -0,0 +1,379 @@
+          public.gen_random_uuid() as etag
+        into new_row;
+        update worker_pools
+        set (
+          provider_id,
+          owner,
+          description,
+          email_on_error,
+          created,
+          last_modified,
+          config,
+          provider_data,
+          previous_provider_ids,
+          etag
+        ) = (
+          new_row.provider_id,
+          new_row.owner,
+          new_row.description,
+          new_row.email_on_error,
+          new_row.created,
+          new_row.last_modified,
+          new_row.config,
+          new_row.provider_data,
+          new_row.previous_provider_ids,
+          new_row.etag
+        )
+        where
+          worker_pools.worker_pool_id = decode_string_key(wmworker_pools_entities_modify.partition_key) and
+          worker_pools.etag = wmworker_pools_entities_modify.old_etag;
+
+        if found then
+          return query select new_row.etag;
+          return;
+        end if;
+
+        perform worker_pools.etag from worker_pools
+        where worker_pools.worker_pool_id = decode_string_key(wmworker_pools_entities_modify.partition_key);
+
+        if found then
+          raise exception 'unsuccessful update' using errcode = 'P0004';
+        else
+          raise exception 'no such row' using errcode = 'P0002';
+        end if;
+      end
+  wmworker_pools_entities_scan:
+    description: See taskcluster-lib-entities
+    mode: read
+    serviceName: worker_manager
+    args: pk text, rk text, condition text, size integer, page integer
+    returns: table (partition_key text, row_key text, value jsonb, version integer, etag uuid)
+    body: |-
+      declare
+      begin
+        if not condition is null then
+          raise exception 'condition not supported';
+        end if;
+        return query
+          select
+            wmworker_pools_entities_scan.partition_key,
+            'workerPool' as row_key,
+            entity_buf_encode(
+              entity_buf_encode(
+                entity_buf_encode(
+                  jsonb_build_object(
+                    'PartitionKey', encode_string_key(worker_pool_id),
+                    'RowKey', 'workerPool',
+                    'workerPoolId', worker_pool_id,
+                    'providerId', provider_id,
+                    'owner', owner,
+                    'description', description,
+                    'emailOnError', email_on_error,
+                    'created', created,
+                    'lastModified', last_modified),
+                  'config', config::text),
+                'providerData', provider_data::text),
+              'previousProviderIds', previous_provider_ids::text) as value,
+            1 as version,
+          worker_pools.etag as etag
+          from worker_pools
+          where
+            partition_key is null or
+            worker_pool_id = decode_string_key(partition_key)
+          order by worker_pool_id
+          limit size + 1
```

Comment:
What would happen if size is not defined?

Reply:
Fortunately tc-lib-entities always sets size, so it doesn't matter (I think it would be `limit NULL` which means no limit, anyway)
https://github.com/taskcluster/taskcluster/blob/54b7a8d276b358a4b3c832a34759c8393d7c6fde/libraries/entities/src/index.js#L463-L467